src/utilities/calculateYearlyEarnings.ts

```diff
@@ -14,23 +18,38 @@ export const calculateYearlyEarningsForAsset = ({ asset }: { asset: Asset }) =>
   const supplyYearlyEarningsCents = assetSupplyBalanceCents.multipliedBy(
     asset.supplyApy.dividedBy(100),
   );
-  // Note that borrowYearlyInterests will always be negative (or 0), since
+  // Note that borrowYearlyEarningsCents will always be negative (or 0), since
   // the borrow APY is expressed with a negative percentage)
-  const borrowYearlyInterestsCents = assetBorrowBalanceCents.multipliedBy(
+  const borrowYearlyEarningsCents = assetBorrowBalanceCents.multipliedBy(
     asset.borrowApy.dividedBy(100),
   );
 
-  return supplyYearlyEarningsCents.plus(borrowYearlyInterestsCents);
+  const yearlyEarningsCents = supplyYearlyEarningsCents.plus(borrowYearlyEarningsCents);
+
+  if (!includeXvs || !asset.xvsSupplyApr.isFinite() || !asset.xvsBorrowApr.isFinite()) {
```

Comment:
I think this is more readable with postive checks, doing the necessary calculations in the if statement and returning `yearlyEarningsCents` if the if statement is not entered

Reply:
I'd rather not. Doing what I'm doing here means we can use a return statement and avoid nesting.